src/base_client.ts

```diff
@@ -81,6 +83,10 @@ export default class BaseClient {
           }
         }
       });
-  });
+    });
+  }
+
+  private sanitisePath(path: string): string {
+    return path.replace(/\/ /g, "/").replace(/\/ $/, "");
```

Comment:
And yeah, I wasn't sure what `normalizePath` did, and was lazy so didn't look it up. It might be that we can just remove it and use the `sanitisePath` stuff, especially seeing as it gives us the specific behaviour we want, e.g. remove any trailing slashes.